### PR TITLE
CIRCSTORE-341 add error handling, add tests

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
- add error handling for processing TLR toggle job;
- create tests;

Resolves: [CIRCSTORE-341](https://issues.folio.org/browse/CIRCSTORE-341)

### CODE REVIEW

File: src/main/java/org/folio/rest/client/ConfigurationClient.java [hunk 1]
```diff
@@ -39,7 +44,18 @@ public Future<TlrSettingsConfiguration> getTlrSettings() {
...
         } else {
-          return succeededFuture(TlrSettingsConfiguration.from(response.bodyAsJsonObject()));
+          try {
+            return objectMapper.readValue(response.bodyAsString(), KvConfigurations.class)
+              .getConfigs().stream()
+              .findFirst()
+              .map(Config::getValue)
+              .map(JsonObject::new)
+              .map(tlrConfig -> succeededFuture(TlrSettingsConfiguration.from(tlrConfig)))
+              .orElse(failedFuture("Failed to find TLR configuration"));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/client/ConfigurationClient.java:54]:
```suggestion
              .orElseGet(() -> failedFuture("Failed to find TLR configuration"));
```

File: src/main/java/org/folio/rest/client/ConfigurationClient.java [hunk 2]
```diff
@@ -39,7 +44,18 @@ public Future<TlrSettingsConfiguration> getTlrSettings() {
...
           return failedFuture(new HttpException(GET, url, response));
         } else {
-          return succeededFuture(TlrSettingsConfiguration.from(response.bodyAsJsonObject()));
+          try {
+            return objectMapper.readValue(response.bodyAsString(), KvConfigurations.class)
+              .getConfigs().stream()
+              .findFirst()
+              .map(Config::getValue)
+              .map(JsonObject::new)
+              .map(tlrConfig -> succeededFuture(TlrSettingsConfiguration.from(tlrConfig)))
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/client/ConfigurationClient.java:53]:
```suggestion
              .map(TlrSettingsConfiguration::from)
              .map(CompletableFuture::succeededFuture)
```

File: src/main/java/org/folio/rest/client/ConfigurationClient.java [hunk 3]
```diff
@@ -39,7 +44,18 @@ public Future<TlrSettingsConfiguration> getTlrSettings() {
...
+              .orElse(failedFuture("Failed to find TLR configuration"));
+          } catch (JsonProcessingException e) {
+            log.error(String.format("Failed to parse response: %s", response.bodyAsString()));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/client/ConfigurationClient.java:56]: Use `{}` placeholders

File: src/main/java/org/folio/service/tlr/TlrFeatureToggleService.java [hunk 4]
```diff
@@ -152,4 +156,16 @@ private Future<Void> refuseWhenJobsInProgressExist() {
         ? succeededFuture()
         : failedFuture(new TlrFeatureToggleJobAlreadyRunningException(jobList)));
   }
+
+  private Future<RowSet<Row>> updateJobAsFailed(TlrFeatureToggleJob job, String errorMessage) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/service/tlr/TlrFeatureToggleService.java:160]: pass Throwable object here

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**